File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        if( k == 1 ) return 0;
        Arrays.sort(nums);
        int min = nums[k-1] - nums[0];
        for(int i = 1; i < nums.length - k + 1 ; i++){
            min = Math.min(min , nums[k+i-1] - nums[i]);
        }
        return min;
    }
    public void printArray(int[] arr){
        for(int i = 0 ; i < arr.length ; i++) System.out.print(arr[i] + ",");
    }
}

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        l, r = 0, k - 1
        min_diff = float('inf')
        while r < len(nums):
            min_diff = min(min_diff, nums[r] - nums[l])
            l, r = l+1, r+1
        return min_diff